### ISSUE TITLE
Apache SSL configuration

### ISSUE THREAD

**Issue by famfop** [Icinga/puppet-icingaweb2#279]
I believe the lines 12 and 25 in the `examples/apache2.pp` need to be adjusted to point to `files/examples/apache2/icingaweb2.conf`. But actually this is not my point. I am trying to configure icingaweb2 resp. apache2 to use SSL on port 443. Using the apache-module, I set my `docroot` to `/usr/share/icingaweb2/public`, i.e. I set the following (in my case $apache_ssl_{crt,key} are set via hiera):
```
  class { 'apache':
    default_vhost => false,
    mpm_module    => 'prefork',
  }
  class { 'apache::mod::rewrite': }

    file { '/etc/ssl':
      ensure => directory,
      owner => 'root',
      group => 'root',
    }

    file { "/etc/ssl/my_website.crt":
      mode => "0644",
      owner => 'root',
      group => 'root',
      content => $apache_ssl_crt,
    }

    file { "/etc/ssl/my_website.key":
      mode => "0644",
      owner => 'root',
      group => 'root',
      content => $apache_ssl_key,
    }

  # Non-ssl vhost
  apache::vhost { 'my_website non-ssl':
    servername => 'my_website',
    port       => '80',
    docroot    => '/usr/share/icingaweb2/public/',
  }

  # SSL vhost at the same domain
  apache::vhost { 'my_website ssl':
    servername => 'my_website',
    port       => '443',
    docroot    => '/usr/share/icingaweb2/public/',
    ssl        => true,
    ssl_cert   => '/etc/ssl/my_website.crt',
    ssl_key   => '/etc/ssl/my_website.key',
  }
```
~The problem: 404 Not Found!~ EDIT: Actually I just found out that this is not the problem. When I open my website and check the content I simply find the content of `/usr/share/icingaweb2/public/index.php` unevaluated. 
So basically my question: ~Is `/usr/share/icingaweb2/public` the correct location or do you know why this fails?~ Why does the php not get evaluated? Or else: Would you recommend a different way of setting up icingaweb2 for https?

**Comment by famfop** [comment]
Ok, I was able to fix this by re-reading the example. It might still be of interest for people using https, so here my working example:
```
  class { 'apache':
    default_vhost => false,
    mpm_module    => 'prefork',
  }
  class { 'apache::mod::php': }
  class { 'apache::mod::rewrite': }

  file {'/etc/apache2/conf.d/icingaweb2.conf':
    source  => 'puppet:///modules/icingaweb2/examples/apache2/icingaweb2.conf',
    require => Class['apache'],
    notify  => Service['apache2'],
  }
  file { '/etc/ssl':
    ensure => directory,
    owner => 'root',
    group => 'root',
  }

  file { "/etc/ssl/myexampledomain.com.crt":
    mode => "0644",
    owner => 'root',
    group => 'root',
    content => $apache_ssl_crt,
  }

  file { "/etc/ssl/myexampledomain.com.key":
    mode => "0644", # stimmt das hier so?
    owner => 'root',
    group => 'root',
    content => $apache_ssl_key,
  }

  # Non-ssl vhost
  apache::vhost { 'myexampledomain.com non-ssl':
    servername => 'myexampledomain.com',
    port       => '80',
    docroot    => '/usr/share/icingaweb2/public/',
  }

  # SSL vhost at the same domain
  apache::vhost { 'myexampledomain.com ssl':
    servername => 'myexampledomain.com',
    port       => '443',
    docroot    => '/usr/share/icingaweb2/public/',
    ssl        => true,
    ssl_cert   => '/etc/ssl/myexampledomain.com.crt',
    ssl_key   => '/etc/ssl/myexampledomain.com.key',
  }

```
Making the site reachable at myexampledomain.com/icingaweb2.
